addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,243 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
+import { useService } from "@web/core/utils/hooks";
+import { useSetupAction } from "@web/webclient/actions/action_hook";
+import { Layout } from "@web/search/layout";
+
+const { Component, useState, onWillStart, markup, onRendered } = owl;
+
+
+export class ProductPricelistReport extends Component {
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+
         this.MAX_QTY = 5;
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
-
-    /**
-     * Add a quantity when add(+) button clicked.
-     *
-     * @private
-     */
-    _onClickAddQty: function () {
+        const pastState = this.props.state || {};
+
+        // Putting everything that needs to be reactive on the useState wrapper.
+
+        // While activeIds and activeModel are not necessarily reactive we need
+        // them to be keeped for the useSetupAction going back and forward on links.
+        this.state = useState({
+            activeIds: pastState.active_ids || this.props.action.context.active_ids,
+            activeModel: pastState.active_model || this.props.action.context.active_model,
+            displayPricelistTitle: pastState.displayPricelistTitle || false,
+            html: "",
+            pricelists: [],
+            quantities: pastState.quantities || [1, 5, 10],
+            selectedPricelist: {},
+        });
+
+        onWillStart(async () => {
+            this.state.pricelists = await this.getPricelists()
+            this.state.selectedPricelist = pastState.selectedPricelist || this.state.pricelists[0];
+
+            this.renderHtml();
+        });
+
+        onRendered(() => {
+            this.env.config.setDisplayName(this.displayName);
+        });
+
+        /*
+        When following the link of a product and coming back we need to keep the 
+        precedent state:
+            - if the pricelist was being showed
+            - wich pricelist is selected at the moment
+            - which quantities
+            - the products (or product variants) for which the report was opened
+            - if the active model is product.product or product.template
+        */
+        useSetupAction({
+            getLocalState: () => {
+                return {
+                    active_ids: this.active_ids,
+                    active_model: this.active_model,
+                    displayPricelistTitle: this.displayPricelistTitle,
+                    quantities: this.quantities,
+                    selectedPricelist: this.selectedPricelist,
+              };
+            },
+        });
+    }
+
+    get activeIds() {
+        return this.state.activeIds;
+    }
+
+    get activeModel() {
+        return this.state.activeModel;
+    }
```

Comment:
I don't feel like getters are that useful for state parameters :/
```suggestion
    get activeIds() {
        return this.state.activeIds;
    }

    get activeModel() {
        return this.state.activeModel;
    }
```